System/Random.hs

```diff
@@ -445,11 +438,18 @@ Minimal complete definition: 'randomR' and 'random'.
 
 -}
 
-class Random a where
-  randomRM :: MonadRandom g m => (a, a) -> g -> m a
 
-  randomM :: MonadRandom g m => g -> m a
+class Uniform a where
```

Comment:
```haskell
(UniformRange a, Bounded a) => Uniform a
```

Is correct, right? Was this left out on purpose?

Reply:
That's right, it is on purpose. @Shimuuar explained why it was not a good idea through a serious of comments, starting here: https://github.com/idontgetoutmuch/random/issues/8#issuecomment-590076154

Reply:
Does @Shimuuar's `X` have a sensible `UniformRange` instance?

Reply:
Not necessary. It doesn't event need to have Ord. Take for example any finite group. Uniform is well defined: every group element appear with equal probability, yet there's no order so there's no UniformR

Reply:
Hm, I'm confused. As I understand it, what you're saying is that

```haskell
Uniform a => UniformRange a
```

does not hold. I agree with that.

But what I am saying is that
```haskell
(Uniform a, Bounded a) => UniformRange a
```

holds. I have not seen a counter-example to this so far, and I don't think there are any. If you agree then I'm in favour of adding this instance.

Reply:
Problem is not that `(UniformRange a, Bounded a) => Uniform a` doesn't hold. It's overly restrictive. There're type where you can define `Uniform` but not `UniformRange`. Examples:

- Finite groups. 
- Tuples. techically you can define `UniformRange`. But that definition require knowing number of elemetns in each tuple type. It's also very inefficient. To sample say `(Word64,Word64,Word64,Word64)` one'll need to work with 256-bit numbers

Reply:
Oh, I see the problem now. What I really meant to write was

```haskell
instance (UniformRange a, Bounded a) => Uniform a where
     uniform = uniformR (minBound, maxBound)
```

That is, I was wondering why that _instance_ is missing. I totally agree that we shouldn't add `(UniformRange a, Bounded a)` as bounds to `Uniform a`.

I see now that I phrased this in a way that suggested I was talking about the latter. Sorry.

Reply:
That will bring us into realm of OverlappingInstances since this instance will overlap with every other instance.

Another possibility is to add default implementation of uniform in terms of uniformRange. But I'm not sure it's worthwhile. For Int/Word{8.16,32,64} it shouldn't be used for performance reasons.

Reply:
:+1: thanks for bearing with me here. I retract my proposal. Just as an idea - people in the future may have similar ideas; it may be worth documenting somewhere all the things that we decided _not_ to do. Not sure where that should go though. Resolving this conversation now. Thanks again!